siwe/siwe.py

```diff
@@ -108,15 +118,35 @@ def __init__(self, message: Union[str, dict] = None, abnf: bool = True):
             message_dict = message
         else:
             raise TypeError
-
         for key in self.__slots__:
             value = message_dict.get(key)
-            if key == "expiration_time" and value is not None:
-                self.expiration_time_parsed = isoparse(value)
-            elif key == "not_before" and value is not None:
-                self.not_before_parsed = isoparse(value)
+
+            if key == "domain" and value == "":
+                raise ValueError("Message `domain` must not be empty")
+            elif key == "address" and value is not None:
+                if not eth_utils.is_checksum_formatted_address(value):
+                    raise ValueError("Message `address` must be in EIP-55 format")
+            elif key == "uri" and value is not None:
+                try:
+                    rfc3987.parse(value, rule='URI')
+                except ValueError:
+                    raise ValueError("Invalid format for field `uri`")
+            elif key == "resources" and value is not None:
+                for url in value:
+                    try:
+                        rfc3987.parse(url, rule='URI')
+                    except ValueError:
+                        raise ValueError("Invalid format for field `resources`")
+
+
             setattr(self, key, value)
 
+    def __str__(self):
```

Comment:
I'm afraid it might be confusing to log a message but have some extra fields that are added in `prepare_message`. Maybe use `__slots__` instead?

Reply:
Decided not to add it to avoid any confusion.